File: Java/19-strong-number-or-not.java
import java.util.Scanner;

class Main {
    public static void main(String[] args) {
        Scanner getInput = new Scanner(System.in);

        System.out.println("Enter a number:");
        int num = getInput.nextInt();
        int temp = num;
        int ans = 0;

        while (num > 0) {
            int digit = num % 10;
            int fact = 1;

            for (int i = 1; i <= digit; i++) {
                fact = fact * i;
            }

            ans += fact;
            num /= 10;
        }

        System.out.println((ans == temp) ? (temp + " is Strong number.") : (temp + " is Not Strong number."));
        getInput.close();
    }
}

File: Python/Important-Programs/19-strong-number-or-not.py
import math

num = int(input("Enter a number: "))
temp = num
ans = 0

while num > 0:
    digit = num%10
    ans += math.factorial(digit)
    num = num//10

if ans == temp:
    print(temp, "is a Strong Number")
else:
    print(temp, "is Not a Strong Number")


# Using for loop for factorial

# num = int(input("Enter a number: "))
# temp = num
# ans = 0

# while num > 0:
#     digit = num%10
#     fact = 1

#     for i in range(1, digit+1):
#         fact = fact * i

#     ans += fact
#     num = num//10

# if ans == temp:
#     print(temp, "is Strong number.")
# else:
#     print(temp, "is Not Strong number.")